python/res/fm/ecl/ecl_config.py

```diff
@@ -0,0 +1,200 @@
+
+
+env_variables = {"F_UFMTENDIAN"    : "big",
+                 "LM_LICENSE_FILE" : "7321@eclipse-lic-no.statoil.no",
+                 "ARCH"            : "x86_64"}
+
+
+intel_path     = "%s/tools/linux_x86_64/intel/mpi/5.0.2.044" % prog_root
+scali_path     = "/opt/scali"
+
+LD_LIBRARY_PATH = os.getenv("LD_LIBRARY_PATH")
+if LD_LIBRARY_PATH:
+    CURRENT_LD_PATH = ":%s" % LD_LIBRARY_PATH
+else:
+    CURRENT_LD_PATH = ""
+
+mpi_config = { "scali" : {"mpirun" : "%s/bin/mpirun" % scali_path   , "env" : {"LD_LIBRARY_PATH" : "%s/lib64%s" % (scali_path , CURRENT_LD_PATH),
+                                                                               "PATH"            : "%s/bin:%s" % (scali_path , os.getenv("PATH"))}},
+               "intel" : {"mpirun" : "%s/bin64/mpirun" % intel_path , "env" : {"I_MPI_ROOT"      : intel_path , 
+                                                                               "P4_RSHCOMMAND"   : "ssh" , 
+                                                                               "LD_LIBRARY_PATH" : "%s/lib64%s"  % (intel_path , CURRENT_LD_PATH),
+                                                                               "PATH"            : "%s/bin64:%s" % (intel_path , os.getenv("PATH"))}}}
+
+def _replace_env_string(value):
+    while True:
+        dollar_pos = value.find("$")
+        if dollar_pos < 0:
+            break
+
+        end_pos = value.find(":", dollar_pos)
+        if end_pos < 0:
+            end_pos = len(value)
+
+        env_var = value[dollar_pos+1:end_pos]
+        env_value = os.getenv(env_var,"")
+        value = value.replace("${}".format(env_var), env_value)
+
+    return value
+
+
+# Small utility function will take a dict as input, and create a new dictionary
+# where all $VARIABLE in the values has been replaced with getenv("VARIABLE").
+
+def _replace_env(env):
+    new_env = {}
+    for key,value in env.items():
+        if "$" in value:
+            new_env[key] = _replace_env_string(value)
+        else:
+            new_env[key] = value
+
+    return new_env
+
+
+class Simulator(object):
+    def __init__(self, name, version, executable, env, mpirun = None):
+        self.name = name
+        self.version = version
+        if not os.access( executable , os.X_OK ):
+            raise OSError("The executable: '{}' can not be executed by user".format(executable))
+
+        self.executable = executable
+        self.env = env
+
+        if mpirun is None:
+            self.mpi = False
+        else:
+            self.mpi = True
+            if not os.access(mpirun, os.X_OK):
+                raise OSError("The mpirun argument: '{}' is not executable by user".format(executable))
+
+        self.mpirun = mpirun
+
+
+
+class EclConfig(object):
+    # The config_file is defined here as a class attribute to allow
+    # for monkey pathching in tests.
+    config_file = os.path.join(os.path.dirname(__file__), "ecl_config.yml")
```

Comment:
I think "for testing purposes" is usually not the best reason for doing something.
What about calling this `DEFAULT_CONFIG_FILE`, and modify `__init__` as follows:
```Python
def __init__(self, config_file=None):
  if config_file is None:
    config_file = self.DEFAULT_CONFIG_FILE
  with open(config_file) as f: 
    ...
```